base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -608,13 +611,7 @@ impl LMDBDatabase {
 
         // Generate PayRef and add to index
         let payref = Self::generate_payment_reference_for_output(header_hash, &output_hash);
-        lmdb_insert(
-            txn,
-            &self.payref_to_output_index,
-            payref.as_slice(),
-            &output_hash,
-            "payref_to_output_index",
-        )?;
+        lmdb_replace(txn, &self.payref_to_output_index, payref.as_slice(), &output_hash, None)?;
```

Comment:
```suggestion
        lmdb_insert(txn, &self.payref_to_output_index, payref.as_slice(), &output_hash, None)?;
```
this will hide errors

Reply:
Ok, let me do something else here, as the issue was collisions because we do not delete payrefs anymore. I will use insert but only report an actual error if the uderlying value is different, which may help us to find the cause of the collisions.
```
- try insert
- if insert fails read the value
  - if the value is the same, log as debug and ignore
  - if the value is different log an error and return an error
  ```